openwisp_notifications/templates/openwisp_notifications/unsubscribe.html

```diff
@@ -0,0 +1,90 @@
+    <title>Manage Subscription Preferences</title>
+    <link rel="stylesheet" type="text/css" href="{% static 'openwisp-notifications/css/unsubscribe.css' %}" />
+  </head>
+  <body>
+    <div class="container">
+      <img
+        src="{% static 'ui/openwisp/images/openwisp-logo-black.svg' %}"
+        alt="OpenWISP Logo"
+        class="logo"
+      />
+      <div class="content">
+        <div class="icon">✉️</div>
+        <h1>Manage Notification Preferences</h1>
+        {% if valid %}
+          <p id="status-message">
+            {% if is_subscribed %}
+              You are currently subscribed to notifications.
+            {% else %}
+              You are currently unsubscribed from notifications.
+            {% endif %}
+          </p>
+          <button id="toggle-btn">
+            {% if is_subscribed %}
+              Unsubscribe
+            {% else %}
+              Subscribe
+            {% endif %}
+          </button>
+          <p id="confirmation-msg" style="display:none;"></p>
+        {% else %}
+          <h1>Invalid or Expired Link</h1>
+          <p>The link you used is invalid or expired. Please contact support.</p>
+        {% endif %}
+        <div class="footer">
+          <p>Manage Other Preferences? <a href="/notifications/preferences">Click here</a></p>
+        </div>
+      </div>
+    </div>
+
+    <script>
+      function updateSubscription(subscribe) {
+        const token = '{{ request.GET.token }}';
+
+        fetch(window.location.href, {
+          method: 'POST',
+          headers: {
+            'Content-Type': 'application/json',
+          },
+          body: JSON.stringify({ subscribe: subscribe })
+        })
+        .then(response => response.json())
+        .then(data => {
+          if (data.success) {
+            const toggleBtn = document.getElementById('toggle-btn');
+            const statusMessage = document.getElementById('status-message');
+            const confirmationMsg = document.getElementById('confirmation-msg');
+
+            if (subscribe) {
+              statusMessage.textContent = "You are currently subscribed to notifications.";
+              toggleBtn.textContent = "Unsubscribe";
+              confirmationMsg.textContent = data.message;
+            } else {
+              statusMessage.textContent = "You are currently unsubscribed from notifications.";
+              toggleBtn.textContent = "Subscribe";
+              confirmationMsg.textContent = data.message;
+            }
+
+            confirmationMsg.style.display = 'block';
+          } else {
+            alert(data.message);
+          }
+        });
+      }
+
+      const toggleBtn = document.getElementById('toggle-btn');
+      toggleBtn.addEventListener('click', function() {
+        const currentStatus = toggleBtn.textContent.trim().toLowerCase();
+        const subscribe = currentStatus === 'subscribe';
+        updateSubscription(subscribe);
+      });
```

Comment:
let's move this to a dedicated file for consistency.

Reply:
Moved!